conans/test/functional/profiles/test_python_profiles.py

```diff
@@ -0,0 +1,99 @@
+import os
+import textwrap
+import unittest
+
+from importlib.util import spec_from_file_location, module_from_spec
+
+from conans.client.tools import environment_append
+from conans.test.utils.tools import TestClient
+
+
+class Profile(dict):
+
+    def __init__(self):
+        self["settings"] = {}
+        self["options"] = {}
+        self["env"] = {}
+
+    @property
+    def settings(self):
+        return self["settings"]
+
+    def update(self, profile):
+        for k, v in profile.items():
+            if k in ["settings", "options", "env"]:
+                self[k].update(v)
+            else:
+                self[k] = v
+
+
+def _load_module_from_file(filename, parent_profile):
+    name = os.path.splitext(os.path.basename(filename))[0]
+    spec = spec_from_file_location(name, filename)
+    config = module_from_spec(spec)
+    for key, value in parent_profile.items():
+        setattr(config, key, value)
+    spec.loader.exec_module(config)
+    return config
+
+
+def _load_profile_python(profile_path, parent_profile=None):
+    # TODO: implement include() of other profiles? variable substitution?
+    parent_profile = parent_profile or Profile()
+    profile_content = _load_module_from_file(profile_path, parent_profile)
+    profile = Profile()
+    for key, value in profile_content.__dict__.items():
+        if not key.startswith("__"):
+            profile[key] = value
+    parent_profile.update(profile)
+    return parent_profile
+
+
+class PythonProfilesTest(unittest.TestCase):
+
+    def test_python_profile_create(self):
+        client = TestClient()
+        pr1_content = textwrap.dedent("""
+            import os
+
+            settings = {
+                 "os": "Windows",
+                 "compiler": "Visual Studio",
+                 "compiler.version": "16",
+                }
+            options = {"shared": True}
+            env = {"pkg:env_var": "bar"}
+            cxxflags = os.environ.get('CXXFLAGS', "")
+            if cxxflags:
+                env['CXXFLAGS'] = cxxflags + " -stdlib=libc++11"
+        """)
+        pr2_content = textwrap.dedent("""
+            from conans import tools
+
+            settings = {"os": "Linux", "build_type": "Release"}
+            env_values = {"cpu_count": str(tools.cpu_count())}
+            if env:
```

Comment:
This reads a bit weird and non idiomatic. Doesn't this profile raise an error if not used together with a parent profile, because ``env`` is an undefined variable?

Reply:
yes, I think so

Reply:
Would you like to rethink the composition proposal yourself, or do you prefer to brainstorm or discuss first?